task-graphs-v2.yaml

```diff
@@ -0,0 +1,370 @@
+      on TileDB Cloud and executed by users who have access to it.
+    type: object
+    properties:
+      uuid:
+        type: string
+        description: A server-assigned unique ID for the UDF, in UUID format.
+      namespace:
+        type: string
+        description: The namespace that owns this task graph log.
+      name:
+        type: string
+        description: >
+          The name of this graph, to appear in URLs.
+          Must be unique per-namespace.
+      readme:
+        type: string
+        description: Documentation for the task graph, in Markdown format.
+      license_id:
+        type: string
+        x-nullable: True
+        description: SPDX license identifier.
+      license_text:
+        type: string
+        x-nullable: True
+        description: Full text of the license.
+      tags:
+        type: array
+        x-nullable: False
+        description: Optional tags to classify the graph.
+        items:
+          type: string
+          description: Unaccented, lowercase, hyphenated Latin characters.
+      nodes:
+        type: array
+        items:
+          $ref: '#/definitions/RegisteredTaskGraphNode'
+        description: >
+          The structure of the graph, in the form of the nodes that make it up.
+          As with `TaskGraphLog`, nodes must topologically sorted, so that any
+          node appears after all the nodes it depends on.
+      # Potential way to deduplicate identical code blocks.
+      # Not in initial version to keep it simpler.
+      # executables:
+      #   type: object
+      #   description: >
+      #     A dictionary of ID to base64-encoded executable code block.
+      #     Used to deduplicate code blocks shared by multiple Nodes.
+
+  RegisteredTaskGraphNode:
+    description: >
+      Information about a single node within a registered task graph.
+      A single node represents one piece of data or a computational step;
+      either as an input value, a data source, or a computation that acts upon
+      earlier nodes. The structure parallels the existing
+      `TaskGraphNodeMetadata`.
+    type: object
+    properties:
+      client_node_id:
+        type: string
+        description: The client-generated UUID of the given graph node.
+      name:
+        type: string
+        x-nullable: true
+        description: >
+          A client-specified name for the node. If provided, this must be
+          unique.
+      depends_on:
+        type: array
+        items:
+          type: string
+        description: >
+          The client_node_uuid of each node that this node depends upon.
+          Used to define the structure of the graph.
+      array_node:
+        $ref: '#/definitions/UDFArrayDetails'
+      input_node:
+        $ref: '#/definitions/TGInputNodeData'
+      sql_node:
+        $ref: '#/definitions/TGSQLNodeData'
+      udf_node:
+        $ref: '#/definitions/TGUDFNodeData'
+
+  TGInputNodeData:
```

Comment:
Outside the scope of this PR but i would like a new input node type for array queries. I imagine it like
```
TGQueryNode:
  namespace: tiledb
  name: dense_array
  query: A[1:10][2:20]
```
Of course this can be simulated using a named udf and a `TGInputNodeData` with the array name but i like it explicitly